src/storage/txn/flow_controller.rs

```diff
@@ -1,1340 +1,83 @@
-// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+// Copyright 2022 TiKV Project Authors. Licensed under Apache-2.0.
 
-// #[PerformanceCriticalPath]
-use std::{
-    cmp::PartialOrd,
-    collections::VecDeque,
-    ops::{Add, AddAssign, Sub, SubAssign},
-    sync::{
-        atomic::{AtomicBool, AtomicU32, Ordering},
-        mpsc::{self, Receiver, RecvTimeoutError, SyncSender},
-        Arc,
-    },
-    thread::{Builder, JoinHandle},
-    time::Duration,
-    u64,
-};
+use std::time::Duration;
 
-use collections::HashMap;
-use engine_rocks::FlowInfo;
-use engine_traits::{CFNamesExt, FlowControlFactorsExt};
-use num_traits::cast::{AsPrimitive, FromPrimitive};
-use rand::Rng;
-use tikv_util::{
-    sys::thread::StdThreadBuildWrapper,
-    time::{Instant, Limiter},
+use super::{
+    singleton_flow_controller::EngineFlowController, tablet_flow_controller::TabletFlowController,
 };
 
-use crate::storage::{config::FlowControlConfig, metrics::*};
-
-const TICK_DURATION: Duration = Duration::from_millis(1000);
-
-const RATIO_SCALE_FACTOR: u32 = 10_000_000;
-const K_INC_SLOWDOWN_RATIO: f64 = 0.8;
-const K_DEC_SLOWDOWN_RATIO: f64 = 1.0 / K_INC_SLOWDOWN_RATIO;
-const MIN_THROTTLE_SPEED: f64 = 16.0 * 1024.0; // 16KB
-const MAX_THROTTLE_SPEED: f64 = 200.0 * 1024.0 * 1024.0; // 200MB
-
-const EMA_FACTOR: f64 = 0.6; // EMA stands for Exponential Moving Average
-
-#[derive(Eq, PartialEq, Debug)]
-enum Trend {
-    Increasing,
-    Decreasing,
-    NoTrend,
+pub enum FlowControlType {
```

Comment:
Why not name it `FlowControl` directly? So initializing it won't need a double calls.